packages/dev/core/src/Engines/thinEngine.ts

```diff
@@ -1941,11 +1941,8 @@ export class ThinEngine extends AbstractEngine {
             webGLPipelineContext.program.__SPECTOR_rebuildProgram = null;
             resetCachedPipeline(webGLPipelineContext);
             if (this._gl) {
+                this._setProgram(null);
```

Comment:
This won't work, because `_setProgram(null)` is doing `if (this._currentProgram !== program)` (with program==null), whereas we want to do `if (this._currentProgram === webGLPipelineContext.program)`.

 You should keep what you did previously and call `this._setProgram()` inside the if block:
```typescript
                if (this._currentProgram === webGLPipelineContext.program) {
                    this._setProgram(null);
                }
```